File: backend/SistemaEmissaoNF.Faturamento.Domain/Entities/NotaFiscal.cs
using SistemaEmissaoNF.Faturamento.Domain.Enums;

namespace SistemaEmissaoNF.Faturamento.Domain.Entities;

public class NotaFiscal
{
    public int Id { get; set; }

    public long NumeroSequencial { get; set; }

    public StatusNotaFiscal Status { get; set; } = StatusNotaFiscal.Aberta;

    public StatusImpressao StatusImpressao { get; set; } = StatusImpressao.Pendente;

    public byte[]? PdfArquivo { get; set; }

    public DateTime? PdfGeradoEm { get; set; }

    public virtual List<NotaFiscalItem> Itens { get; set; } = [];
}

File: frontend/sistema-emissao-nf/src/app/core/models/nota-fiscal.ts
export interface NotaFiscalItem {
  id: number;
  produtoId: number;
  produtoCodigo: number;
  produtoDescricao: string;
  quantidade: number;
}

export interface NotaFiscal {
  id: number;
  numeroSequencial: number;
  numeroFormatado: string;
  status: string;
  statusDescricao: string;
  statusImpressao: string;
  statusImpressaoDescricao: string;
  itens: NotaFiscalItem[];
}

export interface CreateNotaFiscalPayload {
  itens: Array<{
    produtoId: number;
    produtoCodigo: number;
    produtoDescricao: string;
    quantidade: number;
  }>;
}
